### PR TITLE
Fix issue with docker plugin-list install failing boot for docker-based installations

### PR DESCRIPTION
**Pull request by iphoting**
Follows incomplete fix #6009 based on comment <https://github.com/dokku/dokku/pull/6009#issuecomment-1627694934>:
- Fixed `docker/etc/my_init.d/10_dokku_init`
- Added tests in `tests/ci/setup.sh`

### CODE REVIEW

**Comment by iphoting** [comment]
The CI (check-commit) failure looks like a bug in the git checkout code. It looks like it is trying to fetch the PR branch, but the correct repo remote was not added first.

**Comment by josegonzalez** [comment]
Yeah almost seems like a bug in the action? I've seen that happen a few times, I wonder if there is an upstream issue open.

**Comment by iphoting** [comment]
Any reason for this specific line? <https://github.com/dokku/dokku/blob/8e779e25fb527db7cc9c6c867a9f4c228930b738/.github/workflows/ci.yml#L26>

Differs from upstream documentation: https://github.com/actions/checkout/tree/main#Checkout-pull-request-HEAD-commit-instead-of-merge-commit

**Comment by josegonzalez** [comment]
IIRC I got this from [some tutorial](https://monadical.com/posts/filters-github-actions.html), but maybe we can update it to use the sha instead? Not sure if it works as well, but if you want to issue a separate PR with that change, we can test it out :D

**Comment by iphoting** [comment]
> IIRC I got this from [some tutorial](https://monadical.com/posts/filters-github-actions.html), but maybe we can update it to use the sha instead? Not sure if it works as well, but if you want to issue a separate PR with that change, we can test it out :D

That example/tutorial will work for PRs raised from the same repo. For external repos, it will definitely break. For what it's worth, I've created another PR incorporating this change #6035.

**Comment by josegonzalez** [comment]
@iphoting that was merged, mind rebasing?

File: tests/ci/setup.sh [hunk 1]
```diff
@@ -115,6 +115,7 @@ build_dokku_docker_image() {
 
 run_dokku_container() {
   echo "=====> run_dokku_container on CIRCLE_NODE_INDEX: $CIRCLE_NODE_INDEX"
+  echo "redirect: https://github.com/dokku/dokku-redirect.git" >>/var/lib/dokku/plugin-list
```
Comment (josegonzalez) [inline tests/ci/setup.sh:118]: I think maybe we need to create that directory prior to setup? Maybe try creating via `sudo`?

Comment (iphoting) [reply tests/ci/setup.sh:118]: Do you know if sudo can be used in that context?

I'm thinking of doing a `| sudo tee /var/lib/dokku/plugin-list` instead.

Comment (josegonzalez) [reply tests/ci/setup.sh:118]: Yeah, try that.

**Comment by iphoting** [comment]
That's a strange silent failure?

**Comment by josegonzalez** [comment]
Yeah I think you might need to create that directory. Docker will silently create the directory if it doesn't exist. Just `mkdir` it first and then create the file.

**Comment by iphoting** [comment]
> Yeah I think you might need to create that directory. Docker will silently create the directory if it doesn't exist. Just `mkdir` it first and then create the file.

Still fails with a mysterious hash… 🤔

**Comment by iphoting** [comment]
The docker instance startup is still failing:
```
*** Running /etc/my_init.d/10_dokku_init...
       Copying /etc/ssh skel into place
       Copying /home/dokku skel into place
=====> check_container on CIRCLE_NODE_INDEX: 
       Copying /var/lib/dokku/config skel into place
       Copying /var/lib/dokku/data skel into place
       Copying /var/lib/dokku/services skel into place
 !     First argument must be --core or url to plugin
 !     Valid prefixes: https:, git:, ssh:, file:
 !     Valid suffixes: .tar.gz, .tgz
 !     Please retry with valid arguments
*** /etc/my_init.d/10_dokku_init failed with status 1
```

When the image is being build, is the PR’s `/etc/my_init.d/10_dokku_init` version correctly installed?

**Comment by iphoting** [comment]
Resolved. Earlier errors caused by leading space in the URL argument. 😱

**Comment by iphoting** [comment]
May we have this bug fix rolled into a release soon, please?

**Pull request merged**